exchange/bitswap/bitswap.go

```diff
@@ -219,6 +219,15 @@ func (bs *Bitswap) HasBlock(ctx context.Context, blk *blocks.Block) error {
 		return errors.New("bitswap is closed")
 	default:
 	}
+	has, err := bs.blockstore.Has(blk.Key())
+	if err != nil {
+		return err
+	}
+
+	if has {
```

Comment:
move this into `ReceiveMessage`, dont log, but maybe record statistics to be printed out by `ipfs bitswap stat`